CMakeLists.txt

```diff
@@ -30,12 +33,57 @@ endif()
 # Check whether all dependencies are present
 # ---------------------------------------------------------------------------
 
-if (NOT IS_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/ext/robin_map/include")
+if (NB_USE_SUBMODULE_DEPS AND NOT IS_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/ext/robin_map/include")
   message(FATAL_ERROR "The nanobind dependencies are missing! "
     "You probably did not clone the project with --recursive. It is possible to recover "
     "by invoking\n$ git submodule update --init --recursive")
 endif()
 
+# ---------------------------------------------------------------------------
+# Installation rules
+# ---------------------------------------------------------------------------
+
+if(NB_CREATE_INSTALL_RULES AND NOT CMAKE_SKIP_INSTALL_RULES)
+  set(CMAKE_INSTALL_DATAROOTDIR "share"
+    CACHE PATH "Installation path for read-only architecture-independent data root")
+
+  # Normally these would be configurable by the user, but we can't allow that
+  # because the lookup paths are hard-coded in 'cmake/nanobind-config.cmake'
+  set(CMAKE_INSTALL_INCLUDEDIR "${CMAKE_INSTALL_DATAROOTDIR}/nanobind/include")
+  set(NB_INSTALL_SRCDIR "${CMAKE_INSTALL_DATAROOTDIR}/nanobind/src")
+  set(NB_INSTALL_EXTDIR "${CMAKE_INSTALL_DATAROOTDIR}/nanobind/ext")
+  set(NB_INSTALL_CMAKEDIR "${CMAKE_INSTALL_DATAROOTDIR}/nanobind/cmake")
```

Comment:
I guess these could be refactored a bit more to introduce `NB_INSTALL_DATADIR` as the base for these variables and let the user decide where that goes (in case they want to have multiple versions of nanobind `-DNB_INSTALL_DATADIR=share/nanobind-1.8.0`) Thoughts?

```cmake
  set(NB_INSTALL_DATADIR "${CMAKE_INSTALL_DATAROOTDIR}/nanobind"
    CACHE PATH "Installation path for read-only architecture-independent nanobind data files")
    
  # Normally these would be configurable by the user, but we can't allow that
  # because the lookup paths are hard-coded in 'cmake/nanobind-config.cmake'
  set(CMAKE_INSTALL_INCLUDEDIR "${NB_INSTALL_DATADIR}/include")
  set(NB_INSTALL_SRCDIR "${NB_INSTALL_DATADIR}/src")
  set(NB_INSTALL_EXTDIR "${NB_INSTALL_DATADIR}/ext")
  set(NB_INSTALL_CMAKEDIR "${NB_INSTALL_DATADIR}/cmake")
```

Reply:
Added a new commit to include this and also use CMake's [`GNUInstallDirs`](https://cmake.org/cmake/help/latest/module/GNUInstallDirs.html)